tools/schema_loader.cc

```diff
@@ -649,6 +649,89 @@ schema_ptr do_load_schema_from_schema_tables(const db::config& dbcfg, std::files
     }
 }
 
+schema_ptr do_load_schema_from_sstable(const db::config& dbcfg, std::filesystem::path sstable_path, sstring keyspace, sstring table) {
+    if (keyspace.empty()) {
+        keyspace = "my_keyspace";
+    }
+    if (table.empty()) {
+        table = "my_table";
+    }
+
+    db::nop_large_data_handler large_data_handler;
+    gms::feature_service feature_service(gms::feature_config_from_db_config(dbcfg));
+    cache_tracker tracker;
+    sstables::directory_semaphore dir_sem(1);
+    sstables::sstables_manager sst_man("tools::load_schema_from_sstable", large_data_handler, dbcfg, feature_service, tracker,
```

Comment:
All this stuff should be provided from the upper layers and encapsulated in some environment class. I'm sure this is repeated many times. But can be deferred.

Reply:
Currently all the methods in `schema_loader.hh` follow the policy of instantiating all the infrastructure they need themselves, they don't depend on the upper layer for doing this.
The upper layer provides a `db::config` instance, which controls this to a degree.
This was mainly for convenience. Currently the only user of this code is a tool, so I allowed myself to favor convenience over correct layering.

Reply:
We can have a helper function accept a db::config and a function, instantiate all this stuff, and call the function. Just to avoid repetition.